kore/src/Kore/Strategies/Goal.hs

```diff
@@ -106,25 +108,40 @@ transitionRule
 transitionRule = transitionRuleWorker
   where
     transitionRuleWorker CheckProven Proven = empty
-    transitionRuleWorker CheckGoalRem (GoalRem _) = empty
+    transitionRuleWorker CheckGoalRemainder (GoalRemainder _) = empty
+
+    transitionRuleWorker ResetGoalRewritten (GoalRewritten goal) =
+        return (Goal goal)
 
     transitionRuleWorker Simplify (Goal g) =
         Goal <$> simplify g
-    transitionRuleWorker Simplify (GoalRem g) =
-        GoalRem <$> simplify g
+    transitionRuleWorker Simplify (GoalRemainder g) =
+        GoalRemainder <$> simplify g
 
     transitionRuleWorker RemoveDestination (Goal g) =
-        GoalRem <$> removeDestination g
+        Goal <$> removeDestination g
+    transitionRuleWorker RemoveDestination (GoalRemainder g) =
+        GoalRemainder <$> removeDestination g
 
     transitionRuleWorker TriviallyValid (Goal g)
       | isTriviallyValid g = return Proven
-    transitionRuleWorker TriviallyValid (GoalRem g)
+    transitionRuleWorker TriviallyValid (GoalRemainder g)
+      | isTriviallyValid g = return Proven
+    transitionRuleWorker TriviallyValid (GoalRewritten g)
       | isTriviallyValid g = return Proven
 
-    transitionRuleWorker (DerivePar rules) (GoalRem g) =
+    transitionRuleWorker (DerivePar rules) (Goal g) =
+        -- TODO (virgil): Wrap the results in GoalRemainder/GoalRewritten here.
```

Comment:
Is this `TODO` (and the ones below) something you forgot to do now, or does it require more (upcoming) changes in subsequent PR's?

Reply:
More changes in the future. I don't know if it will still make sense after Ana's changes. I added some explanations.